src/Sylius/Bundle/AddressingBundle/Form/Type/ProvinceChoiceType.php

```diff
@@ -40,17 +40,19 @@ public function __construct(RepositoryInterface $repository)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $choiceList = function (Options $options) {
+        $choices = function (Options $options) {
             if (null === $options['country']) {
-                return new ObjectChoiceList($this->repository->findAll(), null, array(), null, 'id');
+                $choices = $this->repository->findAll();
+            } else {
+                $choices = $options['country']->getProvinces();
```

Comment:
Actually, this should be done via ``$resolver->setAllowedTypes(array('country' => CountryInterface::class));``.